### PR TITLE
Provide documentation for the docker build and push github workflow

### LINKED ISSUE #132: Provide documentation for the docker build and push github workflow
This documentation should include:

- [x] Explain the tags that are added to the docker image
- [x] Explain the "how" docker image are pushed to github container registry
- [x] Explain how a kubernetes manifest fetch the image from the github container registry

### PR DESCRIPTION
**Pull request by ThomasCardin**
- Closes #132

### CODE REVIEW

File: docs/gh_docker_workflow.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- **GitHub Action:** A GitHub Action is essentially a workflow or automated process defined within a GitHub repository. It enables users to automate tasks such as building, testing, and deploying software directly within the GitHub platform, providing a powerful way to streamline development workflows.
+
+
+## Explanation of the diagram
+
+The diagram illustrates a workflow for building and pushing an image to the GitHub Container Registry. It outlines the steps involved in this process, including the creation of three specific tags: pull request number, pull request name, and commit SHA based on the commit made in a pull request. Once the image is in the GCR, a Kubernetes deployment can use the image
+
+## Diagram
+
+![Github image workflow](img/gh_docker_workflow.png)
```
Comment (SonOfLope) [inline docs/gh_docker_workflow.md:22]: The information (emails + agency logo) needs to be at bottom of diagram.

**Review by SonOfLope — CHANGES_REQUESTED** [review]

File: docs/gh_docker_workflow.md (hunk 1 above)
Comment (ThomasCardin) [reply docs/gh_docker_workflow.md:22]: Yeah, I took the picture before applying the signature of the CFIA. If it's that important, I will need to remake the diagram from scratch...

Comment (rngadam) [reply docs/gh_docker_workflow.md:22]: the source of the diagram should also be part of the commit to allow future edits

Comment (ThomasCardin) [reply docs/gh_docker_workflow.md:22]: In that case, I think we should'nt use excalidraw and use something else like Mermaid or D2

Comment (ThomasCardin) [reply docs/gh_docker_workflow.md]: Done. The signature is at the bottom of the diagram + I added a backup for the ansible and ghcr_docker_workflow diagram

**Comment by ThomasCardin** [comment]
Note: The markdown issues are fixed inside this pull request: https://github.com/ai-cfia/howard/pull/158

**Review by SonOfLope — APPROVED** [review]

**Pull request merged**